ethcore/client-traits/src/lib.rs

```diff
@@ -423,3 +424,57 @@ pub trait ProvingBlockChainClient: BlockChainClient {
 	/// Get an epoch change signal by block hash.
 	fn epoch_signal(&self, hash: H256) -> Option<Vec<u8>>;
 }
+
+/// Format for importing/exporting blocks
+#[derive(Debug, PartialEq)]
+pub enum DataFormat {
+	Hex,
+	Binary,
+}
+
+impl Default for DataFormat {
+	fn default() -> Self {
+		DataFormat::Binary
+	}
+}
+
+impl FromStr for DataFormat {
+	type Err = String;
+
+	fn from_str(s: &str) -> Result<Self, Self::Err> {
+		match s {
+			"binary" | "bin" => Ok(DataFormat::Binary),
+			"hex" => Ok(DataFormat::Hex),
+			x => Err(format!("Invalid format: {}", x))
+		}
+	}
+}
+
+/// Provides a method for exporting blocks
+pub trait ExportBlocks {
+    /// Export blocks to destination, with the given from, to and format arguments.
+    /// destination could be a file or stdout.
+    /// if the format is hex, each block is written on a new line.
+    /// for binary exports, all block data is written to the same line.
+	fn export_blocks<'a>(
+        &self,
+        destination: Box<dyn std::io::Write + 'a>,
+        from: BlockId,
+        to: BlockId,
+        format: Option<DataFormat>
+    ) -> Result<(), String>;
+}
+
+/// Provides a method for importing blocks
+pub trait ImportBlocks {
```

Comment:
There is an `ImportBlock` trait already which makes this name confusing. I am not 100% clear on the motivation for moving the import/export code from `blockchain` to `client`? For easier testing?

Reply:
Yeah for easier testing. I'm open to an alternative approach, any suggestions?

Reply:
Hmm. One way is to add the trait and tests to `parity/blockchain.rs` (but I guess you tried this already?).

I think part of the confusion arises from "import" being used already for regular import from the network. We could combine the import/export aspect and give emphasis to the fact that this is for importing/exporting data from/to disk. So maybe `pub trait ImportExportToDisk`. It's not poetry but it's pretty clear what its purpose is. 
Another naming scheme could be s/export/dump/ and s/import/load and have `pub trait LoadAndDumpBlocks` with `fn load_blocks_from_disk()` and `dump_blocks_to_disk()`.
Finally, what do you think about moving the dump/load trait to the `ethcore/blockchain` crate where other lower-level functionality lives?